controlplane/internal/controller/clusterdeployment_controller.go

```diff
@@ -273,3 +285,92 @@ func (r *ClusterDeploymentReconciler) createImageRegistry(ctx context.Context, r
+}
+
+// setACICapabilities will set the install config override annotation in the AgentClusterInstall if there
+// are additional enabled capabilities that need to be defined.
+// For SNO (single node openshift) and non-baremetal platform MNO (multi-node openshift), this is set if
+// the OpenshiftAssistedControlPlane has additional capabilities set.
+// For MNO (multi-node openshift), this is set to the default list for baremetal platform,
+// but the OpenshiftAssistedControlPlane can specify additional capabilities to be appended to this list.
+func setACICapabilities(oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane, aci *hiveext.AgentClusterInstall) error {
+	isBaremetalPlatform := aci.Spec.PlatformType == hiveext.BareMetalPlatformType
+	if !capabilitiesSet(oacp.Spec.Config.Capabilities) && !isBaremetalPlatform {
+		return nil
+	}
+	var installCfgOverride InstallConfigOverride
+	baselineCapability, err := getBaselineCapability(oacp.Spec.Config.Capabilities.BaselineCapability, isBaremetalPlatform)
+	if err != nil {
+		return err
+	}
+	installCfgOverride.Capability.BaselineCapability = baselineCapability
+
+	additionalEnabledCapabilities := getAdditionalCapabilities(oacp.Spec.Config.Capabilities.AdditionalEnabledCapabilities, isBaremetalPlatform)
+	installCfgOverride.Capability.AdditionalEnabledCapabilities = additionalEnabledCapabilities
+
+	installCfgOverrideStr, err := json.Marshal(installCfgOverride)
+	if err != nil {
+		return err
+	}
+
+	if aci.Annotations == nil {
+		aci.Annotations = make(map[string]string)
+	}
+
+	aci.Annotations[InstallConfigOverrides] = string(installCfgOverrideStr)
+	return nil
+}
+
+func getBaselineCapability(capability string, isBaremetalPlatform bool) (string, error) {
+	baselineCapability := capability
+	if baselineCapability == "None" || baselineCapability == "vCurrent" {
+		return baselineCapability, nil
+	}
+
+	if baselineCapability == "" {
+		baselineCapability = defaultBaselineCapability
+		if isBaremetalPlatform {
+			baselineCapability = defaultBaremetalBaselineCapability
+		}
+		return baselineCapability, nil
+	}
+
+	var baselineCapabilityRegexp = regexp.MustCompile(`v4\.[0-9]+`)
+	if !baselineCapabilityRegexp.MatchString(baselineCapability) {
+		return "",
+			fmt.Errorf("invalid baseline capability set, must be one of: None, vCurrent, or v4.x. Got: [%s]", baselineCapability)
+	}
+	return baselineCapability, nil
+}
+
+func getAdditionalCapabilities(specifiedAdditionalCapabilities []string, isBaremetalPlatform bool) []string {
+	additionalCapabilitiesList := []string{}
+	if isBaremetalPlatform {
+		additionalCapabilitiesList = append([]string{}, defaultBaremetalAdditionalCapabilities...)
+	}
+
+	for _, capability := range specifiedAdditionalCapabilities {
+		// Ignore MAPI for baremetal MNO clusters and ignore duplicates
+		if (strings.EqualFold(capability, "MachineAPI") && isBaremetalPlatform) || funk.ContainsString(additionalCapabilitiesList, capability) {
```

Comment:
`slices.Contains` is preferable as it doesn't add a dependency on `funk`

Reply:
Changed it, thanks!